### PR TITLE
refactored gamestate to use builderish pattern; moved time conversion…

### PR DESCRIPTION
**Pull request by sam-ao**
… out of gamestate

### CODE REVIEW

File: frontend/src/services/eventHandlers/gameHandler.js [hunk 1]
```diff
@@ -48,15 +54,15 @@ export default class GameHandler {
...
+                    this._convertToSeconds(payload.playTimeSend.duration)
                 );
                 store.commit(this.setGameStateKey, state);
             } else if (
                 currentState.state === WaitingForPlayerToChooseWord.STATE
             ) {
                 const state = new Guessing(
                     payload.playTimeSend.hint,
-                    payload.playTimeSend.duration
+                    this._convertToSeconds(payload.playTimeSend.duration)
```
Comment (enochtsang) [inline frontend/src/services/eventHandlers/gameHandler.js:65]: When calling here, it's not clear what the parameter should be. I think it should be renamed to `convertNanoSecsToSecs

File: frontend/src/modules/scribble/stores/states/gamestates.js [hunk 2]
```diff
@@ -1,71 +1,121 @@
...
-        this.duration = duration / NANOSECOND_TO_SECONDS_FACTOR;
+        this.durationS = durationS;
     }
 }
 
 export class Guessing extends GameState {
     static STATE = "Guessing";
-    constructor(word, duration) {
-        super(Guessing.STATE, true, false, false, true, false);
+    constructor(word, durationS) {
```
Comment (enochtsang) [inline frontend/src/modules/scribble/stores/states/gamestates.js:111]: S seems a bit short to me, Secs/Sec would be better I think.

File: frontend/src/modules/scribble/stores/states/gamestates.js [hunk 3]
```diff
@@ -1,71 +1,121 @@
...
+        super(
+            new GameStateFlags()
+                .forState(Guessing.STATE)
+                .lockedCanvas()
+                .showingWordToGuess()
```
Comment (enochtsang) [inline frontend/src/modules/scribble/stores/states/gamestates.js:116]: This is a cool design pattern, I like it.

File: frontend/src/services/eventHandlers/gameHandler.js [hunk 4]
```diff
@@ -8,12 +8,18 @@ import {
...
 
+const NANOSECOND_TO_SECONDS_FACTOR = 1000000000;
+
 // Event handler for Game API
 export default class GameHandler {
     constructor() {
         this.setGameStateKey = "scribble/setGameState";
     }
 
+    _convertToSeconds(durationNS) {
```
Comment (gordon-cheung) [inline frontend/src/services/eventHandlers/gameHandler.js:19]: for private methods we should be using #, outlined here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#:~:text=Private%20fields%20are%20accessible%20on,only%20static%20methods%20still%20holds.&text=Private%20static%20fields%20are%20added,constructor%20at%20class%20evaluation%20time.

Comment (gordon-cheung) [reply frontend/src/services/eventHandlers/gameHandler.js:19]: Actually, it looks like its experimental and not supported in firefox. Not sure now if we should do this, what do you think?

Comment (sam-ao) [reply frontend/src/services/eventHandlers/gameHandler.js]: I think if its not supported we shouldn't use it, as we do use firefox for testng

**Pull request merged**